tests/unit/lib/batched_queue_tests.py

```diff
@@ -0,0 +1,90 @@
+from datetime import timedelta
+from queue import Queue
+from time import sleep
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+import pytest
+
+from baseplate.lib.batched_queue import BatchedQueue
+
+if TYPE_CHECKING:
+    WorkQueue = Queue[Sequence[int]]
+else:
+    WorkQueue = Queue
+
+
+@pytest.fixture
+def queue() -> WorkQueue:
+    return Queue()
```

Comment:
This seems kinda unnecessary, it's probably clearer to just initialize the Queue() in the test method itself?